api/monitor.py

```diff
@@ -90,10 +101,7 @@ def endpoint(self, timestamp):
         return self.lookup.updates(timestamp)
 
     def run_once(self, start, cutoff):
-        if not self.lookup.authenticated:
-            self.keep_timestamp = False
-            return
-
+        self.assert_authenticated()
```

Comment:
The core branch removed `keep_timestamp`, so this won't work anymore. It's better overall to raise an exception--the exception will be logged as `Timestamp.exception`, and since the Monitor didn't complete its task, `start` and `finish` will not be updated.